edgelet/iotedge/src/check/checks/certificates_quickstart.rs

```diff
@@ -74,40 +74,49 @@ impl CertificatesQuickstart {
-            self.certificate_info = Some(certificate_info);
+        let certificate_info = CertificateValidity::parse(
+            "Device CA certificate".to_owned(),
+            check.device_ca_cert_path.clone().unwrap(),
+        )?;
+        let not_after = certificate_info.not_after;
+        self.certificate_info = Some(certificate_info);
 
-            let now = chrono::Utc::now();
+        let now = chrono::Utc::now();
+        let autogenerated_certs = settings.certificates().device_cert().is_none();
 
+        if autogenerated_certs {
             if not_after < now {
-                return Ok(CheckResult::Warning(
+                Ok(CheckResult::Warning(
                 Context::new(format!(
                     "The Edge device is using self-signed automatically-generated development certificates.\n\
-                     The certs expired at {}. Restart the IoT Edge daemon to generate new development certs.\n\
-                     Please consider using production certificates instead. See https://aka.ms/iotedge-prod-checklist-certs for best practices.",
+                    The certs expired at {}. Restart the IoT Edge daemon to generate new development certs.\n\
+                    Please consider using production certificates instead. See https://aka.ms/iotedge-prod-checklist-certs for best practices.",
                     not_after,
                 ))
                 .into(),
-            ));
+                ))
             } else {
-                return Ok(CheckResult::Warning(
+                Ok(CheckResult::Warning(
                 Context::new(format!(
                     "The Edge device is using self-signed automatically-generated development certificates.\n\
-                     They will expire in {} days (at {}) causing module-to-module and downstream device communication to fail on an active deployment.\n\
-                     After the certs have expired, restarting the IoT Edge daemon will trigger it to generate new development certs.\n\
-                     Please consider using production certificates instead. See https://aka.ms/iotedge-prod-checklist-certs for best practices.",
+                    They will expire in {} days (at {}) causing module-to-module and downstream device communication to fail on an active deployment.\n\
+                    After the certs have expired, restarting the IoT Edge daemon will trigger it to generate new development certs.\n\
+                    Please consider using production certificates instead. See https://aka.ms/iotedge-prod-checklist-certs for best practices.",
                     (not_after - now).num_days(), not_after,
                 ))
                 .into(),
-            ));
+                ))
             }
+        } else if not_after < now {
+            Ok(CheckResult::Failed(
+            Context::new(format!(
+                "The device CA cert has expired at {}. Renew the certificate to restore functionality.",
```

Comment:
This is the critical bit I have added. Let me know if I should change the wording.

Reply:
Some industries might have regulations on certs expiry but I'm not aware of a universal recommendation beyond adapting expiry to deployment/solution specific security profile (e.g. from threat modeling).   Since we don't have the solution specific detail, would it be possible to specify this value through a config.toml parameter?  A default threshold of 90 days sounds reasonable to me.